controllers/cloudstackcluster_controller.go

```diff
@@ -102,43 +101,46 @@ func (r *CloudStackClusterReconciliationRunner) SetReady() (ctrl.Result, error)
 	return ctrl.Result{}, nil
 }
 
-// CheckZoneDetails verifies the Zone CRDs found match against those requested.
-func (r *CloudStackClusterReconciliationRunner) VerifyZoneCRDs() (ctrl.Result, error) {
-	expected := len(r.ReconciliationSubject.Spec.Zones)
-	actual := len(r.Zones.Items)
+// VerifyFailureDomainCRDs verifies the FailureDomains found match against those requested.
+func (r *CloudStackClusterReconciliationRunner) VerifyFailureDomainCRDs() (ctrl.Result, error) {
+	expected := len(r.ReconciliationSubject.Spec.FailureDomains)
+	actual := len(r.FailureDomains.Items)
 	if expected != actual {
-		return r.RequeueWithMessage(fmt.Sprintf("Expected %d Zones, but found %d", expected, actual))
+		return r.RequeueWithMessage(fmt.Sprintf("Expected %d FailureDomains, but found %d", expected, actual))
 	}
-	for _, zone := range r.Zones.Items {
-		if !zone.Status.Ready {
-			return r.RequeueWithMessage(fmt.Sprintf("Zone %s/%s not ready, requeueing.", zone.Namespace, zone.Name))
+	for _, fd := range r.FailureDomains.Items {
+		if !fd.Status.Ready {
+			return r.RequeueWithMessage(fmt.Sprintf("FailureDomains %s/%s not ready, requeueing.", fd.Namespace, fd.Name))
 		}
 	}
 	return ctrl.Result{}, nil
 }
 
-// SetFailureDomains sets failure domains to be used for CAPI machine placement.
-func (r *CloudStackClusterReconciliationRunner) SetFailureDomains() (ctrl.Result, error) {
-	r.ReconciliationSubject.Status.FailureDomains = capiv1.FailureDomains{}
-	for _, zone := range r.Zones.Items {
-		r.ReconciliationSubject.Status.FailureDomains[zone.Spec.ID] = capiv1.FailureDomainSpec{ControlPlane: true}
+// SetFailureDomainsStatusMap sets failure domains in CloudStackCluster status to be used for CAPI machine placement.
+func (r *CloudStackClusterReconciliationRunner) SetFailureDomainsStatusMap() (ctrl.Result, error) {
+	r.ReconciliationSubject.Status.FailureDomains = clusterv1.FailureDomains{}
+	for _, fdSpec := range r.ReconciliationSubject.Spec.FailureDomains {
+		if !strings.HasSuffix(fdSpec.Name, "-"+r.CAPICluster.Name) { // Add cluster name suffix if missing.
+			fdSpec.Name = fdSpec.Name + "-" + r.CAPICluster.Name
```

Comment:
This is the way my IDE preffered the formatting.

I realize it's inconsistent, but it's because one of the strings is an argument in a function.

So, this is already linted to be this way.